settings/src/utilities.js

```diff
@@ -3,14 +3,15 @@ import { store as coreDataStore, useEntityRecord } from '@wordpress/core-data';
 
 /**
  * Fetch the user record.
+ *
+ * @param userId
  */
-export function getUserRecord( userId ) {
-	let userRecord = useEntityRecord( 'root', 'user', userId );
+export function useGetUserRecord( userId ) {
+	const userRecord = useEntityRecord( 'root', 'user', userId );
 
-	// Polyfill in isSaving.
-	if ( undefined === userRecord.isSaving ) {
-		userRecord.isSaving = useSelect( ( select ) => select( coreDataStore ).isSavingEntityRecord( 'root', 'user', userId ) );
-	}
+	userRecord.isSaving = useSelect( ( select ) =>
+		select( coreDataStore ).isSavingEntityRecord( 'root', 'user', userId )
+	);
```

Comment:
Calling hooks conditionally can have unpredictable results, and I'm not seeing any changes in behaviour on the frontend when calling this unconditionally. I believe keeping this value in sync with the store at all times is desirable. @dd32 @iandunn is there a reason this has been written this way? Is it intended to be a perf optimization?